nautobot_data_validation_engine/custom_validators.py

```diff
@@ -146,6 +148,37 @@ def clean(self):
                     continue
                 compliance_class(self.context["object"]).clean()
 
+    def get_compliance_result(self, message=None, instance=None, attribute=None, valid=True):
+        """Generate an DataCompliance object based on the given parameters."""
+        attribute_value = getattr(instance, attribute, None)
+        class_name = f"{instance._meta.app_label.capitalize()}{instance._meta.model_name.capitalize()}CustomValidator"
+
+        # Update `__all__` Data Compliance for given content type and object.
+        if not valid:
+            DataCompliance.objects.filter(
+                compliance_class_name=f"{instance._meta.model_name.capitalize()}DataComplianceRules",
+                content_type=ContentType.objects.get_for_model(instance),
+                object_id=instance.id,
+                validated_attribute="__all__",
+                valid=True,
+            ).update(valid=False)
```

Comment:
Need to update the message text as well:

![image](https://github.com/nautobot/nautobot-app-data-validation-engine/assets/111259311/c7e1aab0-2b9b-4e86-92d9-e8af29a1a2f6)